common/src/main/java/com/wynntils/services/itemfilter/ProfessionItemFilter.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.services.itemfilter;
+
+import com.wynntils.models.items.WynnItem;
+import com.wynntils.models.items.items.game.GatheringToolItem;
+import com.wynntils.models.items.items.game.IngredientItem;
+import com.wynntils.models.items.items.game.MaterialItem;
+import com.wynntils.models.profession.type.ProfessionType;
+
+public class ProfessionItemFilter implements ItemFilter {
+    private ProfessionType profession;
+
+    public ProfessionItemFilter(ProfessionType profession) {
+        // TODO FILTER NULL
+        this.profession = profession;
+    }
+
+    @Override
+    public boolean matches(WynnItem wynnItem) {
+        if (profession == null) {
+            return false;
+        } else if (wynnItem instanceof IngredientItem ingredient) {
+            return ingredient.getIngredientInfo().professions().contains(profession);
+        } else if (wynnItem instanceof MaterialItem materialItem) {
+            return materialItem
+                            .getMaterialProfile()
+                            .getResourceType()
+                            .getMaterialType()
+                            .getProfessionType()
+                    == profession;
+        } else if (wynnItem instanceof GatheringToolItem gatheringToolItem) {
+            return gatheringToolItem.getToolProfile().toolType().getProfessionType() == profession;
+        }
```

Comment:
We kind of need to add `ProfessionItemProperty`. If you want to, go ahead. Otherwise, this is fine and we will add it later.

Reply:
I think it is better to wait the end of this infinitely long PR because of all my mistakes, before focusing on adding more filters and improving this one.

Reply:
I agree, that sounds like a reasonable extension. I can help do it in a separate PR as well, push it to main, and then you can merge it and use it. Ping me if you want that to happen. (I only check this PR once in a while otherwise).

Reply:
@magicus It would be great. It will probably take you far less time than me to properly implement it 😄. Thank you!

Reply:
I totally forgot about this! Sorry! 😊 Here is the PR: https://github.com/Wynntils/Artemis/pull/2015